### PR TITLE
Add missing libllvm headers

### PR DESCRIPTION
**Pull request by phate**
1. Add several headers that were missing in the Makefile
2. Sort header files alphabetically

### CODE REVIEW

File: jlm/llvm/Makefile.sub [hunk 1]
```diff
@@ -65,70 +65,77 @@ libllvm_SOURCES = \
     jlm/llvm/opt/unroll.cpp \
 
 libllvm_HEADERS = \
```
Comment (phate) [inline jlm/llvm/Makefile.sub:67]: Holding this PR back until @haved merged the dot backend as it touches the files in the Makefile. I rather deal with the conflict here.

**Comment by phate** [comment]
@haved @sjalander This PR is ready for review.

**Review by sjalander — COMMENTED** [review]
I haven't thought this through but could we add a CI test that checks for this.
Ideally, it should check that all listed .hpp files exist and that none is missing.

File: jlm/llvm/Makefile.sub [hunk 2]
```diff
@@ -66,71 +66,78 @@ libllvm_SOURCES = \
...
-	jlm/llvm/ir/operators/lambda.hpp \
-	jlm/llvm/ir/aggregation.hpp \
-	jlm/llvm/backend/dot/DotWriter.hpp \
-	jlm/llvm/backend/rvsdg2jlm/rvsdg2jlm.hpp \
-	jlm/llvm/backend/rvsdg2jlm/context.hpp \
-	jlm/llvm/backend/jlm2llvm/jlm2llvm.hpp \
-	jlm/llvm/backend/jlm2llvm/type.hpp \
-	jlm/llvm/backend/jlm2llvm/instruction.hpp \
-	jlm/llvm/backend/jlm2llvm/context.hpp \
+    jlm/llvm/backend/dot/DotWriter.hpp \
```
Comment (sjalander) [inline jlm/llvm/Makefile.sub:69]: Do we need to list them individually or could we simply use
jlm/llvm/backend/dot/*.hpp

Comment (phate) [reply jlm/llvm/Makefile.sub:69]: I do not know. @caleridas Could you help me out here, please?

**Comment by phate** [comment]
@sjalander I created issue #617 for the CI request.

File: jlm/llvm/Makefile.sub (hunk 2 above)
Comment (phate) [reply jlm/llvm/Makefile.sub:69]: Neither I am unsure whether using wildcards here would be a good idea....

Comment (caleridas) [reply jlm/llvm/Makefile.sub:69]: would prefer to list them explicitly, and certainly a build rule to check for consistency would be in order -- can take a look at that

using wildcards... instinctively I would prefer not to as it also tends to cause problems with spurious dependencies e.g. with files accidentally left in the build tree -- also, when "installing" (=exporting to other projects) files, it is essential that the correct files are selected

**Comment by phate** [comment]
I will close this PR as it was already handled in #619

**Pull request closed without merging**